test/server/dispatcher.test.ts

```diff
@@ -818,3 +818,51 @@ describe("given a request that contains the OpenApi basePath", () => {
     expect(response.body).toBe("ok");
   });
 });
+
+describe("given a request that contains the differently cased path", () => {
+  it("correctly returns the desired path response even when case of path does not match", async () => {
+    const registry = new Registry();
+
+    registry.add("/abc", {
+      POST() {
+        return {
+          body: "ok",
+          status: 200,
+        };
+      },
+    });
+
+    const dispatcher = new Dispatcher(registry, new ContextRegistry(), {
+      paths: {
+        "/Abc": {
+          post: {
+            responses: {
+              200: {
+                content: {
+                  "text/plain": {
+                    schema: {
+                      type: "string",
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    });
+
+    const response = await dispatcher.request({
+      body: "",
+      headers: {},
+      method: "POST",
+      path: "/abc",
+      query: {},
+      req: { path: "/abc" },
+    });
+
+    expect(response.status).toBe(200);
+
+    expect(response.body).toBe("ok");
```

Comment:
Let's add assertions where the path is "ABC" as well. So that we can be protected against accidentally dropping a `toLowerCase()` at some point in the future.

Reply:
done